android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/RpcClient.java

```diff
@@ -280,7 +280,7 @@ public synchronized boolean authorize(String password) {
 	 * @return true if connected to BOINC core client, false if not connected
 	 */
 	public final boolean isConnected() {
-		return (mSocket != null) ? mSocket.isConnected() : false;
+		return (mSocket != null) && mSocket.isConnected();
```

Comment:
*parentheses*
 
Also, I think you can remove the parentheses around `(mSocket != null)` when you add the outer ones.

Reply:
Fixed.